src/iop/expfusion.c

```diff
@@ -0,0 +1,1268 @@
+          = 4.f * img_src[((i + 1) * wd + (j)) * ch + k];
+
+  // bottom row
+  i = ht;
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(img_src, img_tmp2, wd, ch, padding, wd_upsampled, i) \
+  schedule(static)
+#endif
+  for(int j = 0; j < wd; j++)
+    for(int k = 0; k < ch; k++)
+      img_tmp2[((2 * (i + padding)) * wd_upsampled + (2 * (j + padding))) * ch + k]
+          = 4.f * img_src[((i - 1) * wd + (j)) * ch + k];
+
+  // left edge
+  int j = -1;
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(img_src, img_tmp2, ht, wd, ch, padding, wd_upsampled, j) \
+  schedule(static)
+#endif
+  for(int ii = 0; ii < ht; ii++)
+    for(int k = 0; k < ch; k++)
+      img_tmp2[((2 * (ii + padding)) * wd_upsampled + (2 * (j + padding))) * ch + k]
+          = 4.f * img_src[((ii)*wd + (j + 1)) * ch + k];
+
+  // right edge
+  j = wd;
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(img_src, img_tmp2, ht, wd, ch, padding, wd_upsampled, j) \
+  schedule(static)
+#endif
+  for(int ii = 0; ii < ht; ii++)
+    for(int k = 0; k < ch; k++)
+      img_tmp2[((2 * (ii + padding)) * wd_upsampled + (2 * (j + padding))) * ch + k]
+          = 4.f * img_src[((ii)*wd + (j - 1)) * ch + k];
+
+  // corners
+  for(int k = 0; k < ch; k++)
+  {
+    i = -1;
+    j = -1;
+    img_tmp2[((2 * (i + padding)) * wd_upsampled + (2 * (j + padding))) * ch + k]
+        = 4.f * img_src[((i + 1) * wd + (j + 1)) * ch + k];
+    j = wd;
+    img_tmp2[((2 * (i + padding)) * wd_upsampled + (2 * (j + padding))) * ch + k]
+        = 4.f * img_src[((i + 1) * wd + (j - 1)) * ch + k];
+    i = ht;
+    j = -1;
+    img_tmp2[((2 * (i + padding)) * wd_upsampled + (2 * (j + padding))) * ch + k]
+        = 4.f * img_src[((i - 1) * wd + (j + 1)) * ch + k];
+    j = wd;
+    img_tmp2[((2 * (i + padding)) * wd_upsampled + (2 * (j + padding))) * ch + k]
+        = 4.f * img_src[((i - 1) * wd + (j - 1)) * ch + k];
+  }
+
+  // blur
+  _convolve_replicate(img_tmp2, wd_upsampled, ht_upsampled, ch, filter, filter, img_tmp1, img_tmp2);
+
+  // remove the border and copy result
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(img_dest, img_tmp2, up_ht, up_wd, ch, wd_upsampled) \
+  schedule(static) \
+  collapse(2)
+#endif
+  for(int ii = 0; ii < up_ht; ii++)
+    for(int jj = 0; jj < up_wd; jj++)
+      for(int k = 0; k < ch; k++)
+        img_dest[(ii * up_wd + jj) * ch + k] = img_tmp2[((ii + 2) * wd_upsampled + (jj + 2)) * ch + k];
+}
+
+static void _build_gaussian_pyramid(const float *const img_src, const size_t wd, const size_t ht, const int ch,
+                                    dt_image_pyramid_t *const img_tmp1, dt_image_pyramid_t *const img_tmp2,
+                                    dt_pyramid_t *const pyramid_dest)
+{
+  // copy image to the finest level
+  _image_copy(img_src, wd, ht, ch, pyramid_dest->images[0].img);
+
+  const float a = 0.4;
```

Comment:
Is there a reason you're using a different value for a than enfuse here?  Even more specifically, why does this differ from the value of a used in reconstruct_laplacian?

Reply:
This is the value used in LocalLaplacian, it gives good results. The one in the basecurve doesn't.
Again, I didn't check enfuse, but that's easy to try, provided that you know where the values are.

Reply:
Interesting.  When I played with it, it didn't have significant changes.  One observation still is that this doesn't match the values used when you collapse the pyramid - that's using 0.375

Enfuse implementation is at https://github.com/jackmitch/enblend-enfuse/blob/master/src/pyramid.h - if you look at the comments, they're using a pattern prior to normalization if [1, 4, 6, 4, 1] - which corresponds to a value for a of 0.375.  The [1,4,6,4,1] pattern is what the exposure fusion module used too.